File: ws/sisalema/u2/ws22UnitTesting/BasicOperations/src/test/java/ec/edu/espe/basicoperations/controller/BasicOperationTest.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/UnitTests/JUnit5TestClass.java to edit this template
 */
package ec.edu.espe.basicoperations.controller;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author Yostin Sisalema, Code Masters, DCCO - ESPE
 */
public class BasicOperationTest {
    
    public BasicOperationTest() {
    }

    /**
     * Test of addTwoNumbers method, of class BasicOperation.
     */
    @Test
    public void testAddTwoNumbers() {
        System.out.println("addTwoNumbers");
        float addend1 = 1.2F;
        float addend2 = 2.4F;
        float expResult = 3.6F;
        float result = BasicOperation.addTwoNumbers(addend1, addend2);
        //assertEquals(expResult, result, 0);
        assertEquals(expResult, result, 0.0001);
    }
    
    public void testAddTwoNumbersOneNegative() {
        System.out.println("addTwoNumbers");
        float addend1 = 1.8F;
        float addend2 = -2.2F;
        float expResult = -0.4F;
        float result = BasicOperation.addTwoNumbers(addend1, addend2);
        //assertEquals(expResult, result, 0);
        assertEquals(expResult, result, 0.0001);
    }
        public void testAddTwoNumbersIntegers() {
        System.out.println("addTwoNumbers");
        float addend1 = 1;
        float addend2 = 2;
        float expResult = 3;
        float result = BasicOperation.addTwoNumbers(addend1, addend2);
        //assertEquals(expResult, result, 0);
        assertEquals(expResult, result, 0.0001);
    }
}

File: hw/alvarado/u2/hw13ToOurProgrammingLanguage/BasicOperation.test.js
const BasicOperation = require('./BasicOperation');

test('addTwoNumbers with positive numbers', () => {
    const addend1 = 1.2;
    const addend2 = 2.4;
    const expected = 3.6;
    const result = BasicOperation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(expected, 5);
  });
  
  test('addTwoNumbers with positive and negative numbers', () => {
    const addend1 = 1.8;
    const addend2 = -2.2;
    const expected = -0.4;
    const result = BasicOperation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(expected, 5);
  });
  
  test('addTwoNumbers with negative numbers', () => {
    const addend1 = -1.5;
    const addend2 = -2.5;
    const expected = -4.0;
    const result = BasicOperation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(expected, 5);
  });
  
  test('addTwoNumbers with zero', () => {
    const addend1 = 0;
    const addend2 = 0;
    const expected = 0;
    const result = BasicOperation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(expected, 5);
  });
  
  test('addTwoNumbers with large numbers', () => {
    const addend1 = 1000000;
    const addend2 = 2000000;
    const expected = 3000000;
    const result = BasicOperation.addTwoNumbers(addend1, addend2);
    expect(result).toBe(expected);
  });